packages/core/src/renderer/components/tree-view/tree-view.tsx

```diff
@@ -3,103 +3,131 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
-import "./tree-view.scss";
-
-import React, { useEffect, useRef } from "react";
-import { Icon } from "../icon";
-import TreeView from "@material-ui/lab/TreeView";
-import TreeItem from "@material-ui/lab/TreeItem";
+import styles from "./tree-view.module.scss";
+import type { MouseEventHandler } from "react";
+import React, { useState } from "react";
 import { cssNames } from "../../utils";
+import { Icon } from "../icon";
 
-import _ from "lodash";
-import getDeepDash from "deepdash";
-
-const deepDash = getDeepDash(_);
-
-export interface NavigationTree {
-  id: string;
-  parentId?: string;
-  name: string;
-  selected?: boolean;
-  children?: NavigationTree[];
+export interface TreeViewClasses {
+  root?: string;
 }
 
-export interface RecursiveTreeViewProps {
-  data: NavigationTree[];
+export interface TreeViewProps {
+  classes?: TreeViewClasses;
+  children: JSX.Element[] | JSX.Element;
```

Comment:
```suggestion
  children: React.ReactNode;
```

_From `lens/node_modules/@types/react/index.d.ts`_:

```tsx
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
    type ReactChild = ReactElement | ReactText;
    type ReactFragment = {} | Iterable<ReactNode>;
    type ReactText = string | number;
```